src/dmd/root/file.d

```diff
@@ -309,10 +299,20 @@ nothrow:
 
     /* Set buffer
      */
-    extern (C++) void setbuffer(void* buffer, size_t len)
+    extern (C++) void setbuffer(void* buffer, size_t len, bool transferOwnership = false)
     {
         this.buffer = cast(ubyte*)buffer;
         this.len = len;
+        this._ref = !transferOwnership;
```

Comment:
All these negations and their defaults hurt my brain. There's something very simple going on here, and it needs to be much more simply expressed.